include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -75,111 +46,143 @@ class CComputeBlit : public core::IReferenceCounted
 		m_dummySSBO = device->createDeviceLocalGPUBufferOnDedMem(creationParams, 1ull);
 	}
 
-	void getDefaultDescriptorSetLayouts(core::smart_refctd_ptr<video::IGPUDescriptorSetLayout>& outBlitDSLayout,
-		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout>& outKernelWeightsDSLayout, const asset::IBlitUtilities::E_ALPHA_SEMANTIC alphaSemantic)
+	void getDefaultDescriptorSetLayouts(core::smart_refctd_ptr<video::IGPUDescriptorSetLayout>* outBlitDSLayout,
+		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout>* outKernelWeightsDSLayout, const asset::IBlitUtilities::E_ALPHA_SEMANTIC alphaSemantic)
 	{
-		if (alphaSemantic == asset::IBlitUtilities::EAS_REFERENCE_OR_COVERAGE)
+		if (outBlitDSLayout)
 		{
-			if (!m_blitDSLayouts[1])
-			{
-				constexpr uint32_t BLIT_DESCRIPTOR_COUNT = 3u;
-				asset::E_DESCRIPTOR_TYPE types[BLIT_DESCRIPTOR_COUNT] = { asset::EDT_COMBINED_IMAGE_SAMPLER, asset::EDT_STORAGE_IMAGE, asset::EDT_STORAGE_BUFFER }; // input image, output image, alpha statistics
-				m_blitDSLayouts[1] = getDSLayout(BLIT_DESCRIPTOR_COUNT, types, device.get(), sampler);
-			}
+			constexpr auto MAX_BLIT_DESCRIPTOR_COUNT = 3;
+			asset::E_DESCRIPTOR_TYPE types[MAX_BLIT_DESCRIPTOR_COUNT] = { asset::EDT_COMBINED_IMAGE_SAMPLER, asset::EDT_STORAGE_IMAGE, asset::EDT_STORAGE_BUFFER }; // input image, output image, alpha statistics
+
+			const auto blitType = (alphaSemantic == asset::IBlitUtilities::EAS_REFERENCE_OR_COVERAGE) ? EBT_COVERAGE_ADJUSTMENT : EBT_REGULAR;
+			const auto actualDescriptorCount = (blitType == EBT_COVERAGE_ADJUSTMENT) ? 3 : 2;
 
-			outBlitDSLayout = m_blitDSLayouts[1];
+			if (!m_blitDSLayout[blitType])
+				m_blitDSLayout[blitType] = getDSLayout(actualDescriptorCount, types, device.get(), sampler);
```

Comment:
do this in constructor

Reply:
Are you sure?

The reason I removed it is because you may not need all the types of `m_blitDSLayouts`. When it is needed, it will be created here i.e. on demand.

Reply:
you only need 2, easier to make and construct up-front

Reply:
Resolved.